pex/bin/pex.py

```diff
@@ -175,6 +175,18 @@ def configure_clp_pex_resolution(parser):
         help="Additional cheeseshop indices to use to satisfy requirements.",
     )
 
+    parser.add_argument(
+        "--pex-repository",
+        dest="pex_repository",
+        metavar="FILE",
+        default=None,
+        type=str,
```

Comment:
Can you envision wanting to support multiple PEX repositories in the future?

Reply:
Absolutely not - that would be incorrect. The only reason it's safe to resolve from a single `--pex-repository` is because it contains the result of a single Pip resolve that took into account all the constraints at once. The minute you mix in a second source - be it another PEX repository, and --index or a --find-links repo, you force PEX to produce invalid results or else learn a full implementation of a resolve algorithm. The former is clearly not acceptable and the latter would pretty much fully negate the benefit of embedding Pip and delegating to it to do real resolves.

Reply:
One way to look at it is Pex uniformly only supports a single Pip resolve. In a normal resolve this is done by Pip which happens to take in potentially multiple sources to run its singular logic against. In a `--pex-repository` resolve its done by Pex, but its not a real resolve, its just Pex picking the subset of the single resolve Pip did at some point in the past to build the PEX repository.